com.palantir.typescript/src/com/palantir/typescript/text/reconciler/PresentationReconciler.java

```diff
@@ -118,17 +118,22 @@ public IPresentationRepairer getRepairer(String contentType) {
      */
     private void processEvent(TextEvent event) {
         IRegion damagedRegion = this.getDamagedRegion(event);
-        EndOfLineState lastDamagedLexState = this.updateFinalLexStates(event, damagedRegion);
-        TextPresentation presentation = this.createPresentation(damagedRegion, lastDamagedLexState);
+        if(damagedRegion != null) {
+            EndOfLineState lastDamagedLexState = this.updateFinalLexStates(event, damagedRegion);
+            TextPresentation presentation = this.createPresentation(damagedRegion, lastDamagedLexState);
 
-        this.viewer.changeTextPresentation(presentation, false);
+            this.viewer.changeTextPresentation(presentation, false);
+        }
     }
 
     /**
      * Gets the damaged region by selecting the whole lines touched by the text edit.
      */
     private IRegion getDamagedRegion(TextEvent event) {
         IDocument document = this.viewer.getDocument();
+        if(document == null) {
+            return null;
```

Comment:
guava's be fine (thats what everyone uses before JDK 1.8), but yea not really important :P